packages/datadog-instrumentations/src/helpers/instrument.js

```diff
@@ -90,32 +90,31 @@ function getBasedir (id) {
 }
 
 if (semver.satisfies(process.versions.node, '>=16.0.0')) {
-  exports.bind = AsyncResource.bind
-  exports.bindAsyncResource = AsyncResource.prototype.bind
+  exports.AsyncResource = AsyncResource
 } else {
-  exports.bindAsyncResource = function bindAsyncResource (fn, thisArg) {
-    thisArg = thisArg || this
-    const ret = this.runInAsyncScope.bind(this, fn, thisArg)
-    Object.defineProperties(ret, {
-      'length': {
-        configurable: true,
-        enumerable: false,
-        value: fn.length,
-        writable: false
-      },
-      'asyncResource': {
-        configurable: true,
-        enumerable: true,
-        value: this,
-        writable: true
-      }
-    })
-    return ret
-  }
+  exports.AsyncResource = class extends AsyncResource {
+    static bind (fn, type, thisArg) {
+      type = type || fn.name
+      return (new exports.AsyncResource(type || 'bound-anonymous-fn')).bind(fn, thisArg)
```

Comment:
Wanna just istanbul ignore the untested parts of this and add a comment saying why?

Reply:
This entire file is not tested at all as it's tested implicitly by all the plugins.

Reply:
Yeah and if there are parts that are untested, then we should either istanbul ignore them or test them. In this case, I don't think it's worth testing because this is in Node.js core and tested there.

Reply:
Could this be a bug? All the lines that are supposedly not covered are used. It's also worth noting that the old code which was identical but without the class had the same issue.

Reply:
I don't know what the problem was so I ended up simply adding tests.